controllers/llamastackdistribution_controller.go

```diff
@@ -257,6 +316,20 @@ func (r *LlamaStackDistributionReconciler) reconcileDeployment(ctx context.Conte
 	// Configure storage
 	podSpec := configurePodStorage(instance, container)
 
+	// Prepare annotations for the pod template
+	podAnnotations := make(map[string]string)
+
+	// Add ConfigMap hash to trigger restarts when the ConfigMap changes
+	if instance.Spec.Server.UserConfig != nil && instance.Spec.Server.UserConfig.ConfigMapName != "" {
+		configMapHash, err := r.getConfigMapHash(ctx, instance)
+		if err != nil {
+			logger.Error(err, "failed to get ConfigMap hash")
+			// Don't fail the reconciliation, but log the error
```

Comment:
Why not fail the reconciliation and requeue? If we don't apply the hash, we cannot get a restart, right?

Reply:
Updated code to fail reconciliation in this case